src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -0,0 +1,284 @@
+            self._sas_admin.InjectPalDatabaseRecord(config['pals'][index])
+            ppa_ids.append(self._sas_admin.InjectZoneData({'record': ppa}))          
+        # inject N3 Esc sensor
+        for esc_sensor in config['esc_sensors']:
+            self._sas_admin.InjectEscSensorDataRecord({'record': esc_sensor})   
+            
+        # STEP 3
+        self._sas_admin.TriggerFullActivityDump()
+        # STEP 4
+        response = self._sas.GetFullActivityDump()
+        # STEP 5
+        # check dump message format
+        self.assertContainsRequiredFields("FullActivityDump.schema.json", response)
+        # an array for each record type
+        cbsd_dump_data = []
+        ppa_dump_data = []
+        esc_sensor_dump_data = []
+        # STEP 5
+        self.assertContainsRequiredFields("FullActivityDump.schema.json", response)
+        
+        # step 6 and check   
+        # download dump files and fill corresponding arrays
+        for dump_file in response['files']:
+            self.assertContainsRequiredFields("ActivityDumpFile.schema.json",
+                                               dump_file[0])
+            if dump_file['recordType'] ==  'cbsd':
+                cbsd_dump_data.append(self._sas.DownloadFile(dump_file['url'])['recordData'])   
+            elif dump_file['recordType'] ==  'esc_sensor':
+                esc_sensor_dump_data.append(self._sas.DownloadFile(dump_file['url'])['recordData'])
+            elif dump_file['recordType'] ==  'zone':
+                ppa_dump_data.append(self._sas.DownloadFile(dump_file['url'])['recordData'])
+        
+        # verify the length of records equal to the inserted ones
+        self.assertEqual(len(config['registrationRequests']), len(cbsd_dump_data))
+        self.assertEqual(len(config['ppas']), len(ppa_dump_data))
+        self.assertEqual(len(config['esc_sensors']), len(esc_sensor_dump_data))
+        
+        # verify the schema of record and that the injected ppas exist in the dump files
+        for index, ppa in enumerate(config['ppas']):
+            ppa_record = [record for record in ppa_dump_data if record['id'].split("/")[-1] == ppa_ids[index]][0]
+            del ppa_record['id']
+            self.assertContainsRequiredFields("zoneData.schema.json", ppa_record)
+            self.assertDictEqual(ppa, ppa_record)
+            
+        # verify the schema of record and that the injected esc sensors exist in the dump files        
+        for esc in config['esc_sensors']:
+            esc_record = [record for record in ppa_dump_data if record['id'] == esc['id']][0]
+            self.assertContainsRequiredFields("EscSensorRecord.schema.json", esc_record)
+            self.assertDictEqual(esc, esc_record)
+            
+        # verify that retrieved cbsd dump files have correct schema
+        for cbsd in cbsd_dump_data:
+            self.assertContainsRequiredFields("CbsdData.schema.json", record)   
```

Comment:
Sorry, I missed your question about banned content. After looking at the list of optional fields in the SAS-SAS spec and R2-ISC-{06, 07}, it seems like only cbsdInfo is banned. I had remembered the list being much longer, though, so it might be worth consulting with... WG5? WG1? I know it's the operators who care most about which information is exchanged.

I thought userId was banned from exchange but it seems to be allowed based on R2-ISC-06(b)(i).

One question is what should happen to cpiSignatureData, etc. if a CPI is used to register the device. Do we include the registration request exactly as we received it from the CBSD, or do we move the CPI's information to installationParam (plaintext) instead? The latter seems simpler but I don't know that anyone has discussed it.

Reply:
in the code i assumed the later proposal, Christophe sent an email to SAS admins group asking about this point so we can agree(I hope :) ) on the content of the CBSD record.

Reply:
Thanks to both of you for starting that discussion. I replied with the latter proposal and Masoud from FW indicated agreement, so that seems like a safe direction to go with this code. We'll probably be able to confirm the decision next week and I don't think that decision needs to block the submission of this PR.